### PR TITLE
doc: clarifies contracts of @DerivedProperty

### PR DESCRIPTION
**Pull request by monperrus**
per our discussion on #1586,  PropertyGetter can be used for @DerivedProperty as well

### CODE REVIEW

**Comment by pvojtechovsky** [comment]
I do not understand purpose of this change. But I do not need to know everything. I just cannot review this PR. I personaly would keep this annotation in the model. It had a meening

File: src/main/java/spoon/support/reflect/code/CtInvocationImpl.java [hunk 1]
```diff
@@ -175,6 +176,8 @@ public void replace(CtStatement element) {
 		if (type != null) {
 			type.setParent(this);
 		}
+		// even if the setter delegates to getExecutable, it makes sense to create an event
+		getFactory().getEnvironment().getModelChangeListener().onObjectUpdate(this, TYPE, type, getExecutable().getType());
```
Comment (pvojtechovsky) [inline src/main/java/spoon/support/reflect/code/CtInvocationImpl.java]: So model change listener will be called twice for single modification? I do not like it. 
@monperrus Could you explain why it is good?

Comment (monperrus) [reply src/main/java/spoon/support/reflect/code/CtInvocationImpl.java]: you're right, it's not good.

**Comment by monperrus** [comment]
The most convincing reason to keep @DerivedProperty is in ASTChecker with this contract.

Contract: a setter with @DerivedProperty never triggers a model intercession event.

So let's make this PR only a clarification one.

File: src/main/java/spoon/support/DerivedProperty.java [hunk 2]
```diff
@@ -23,10 +23,16 @@
...
 import java.lang.annotation.Target;
 
-/** Tells that a metamodel property is derived, ie computed from the value of another property.
+/**
+ * Tells that a metamodel property is derived, ie computed from the value of another property.
+ *
  * For instance {@link CtType#getFields()}  is derived from {@link CtType#getTypeMembers()}
  *
- * This annotation is used for specifying CtScanner: derived properties are never scanned.
+ * It can be put on getter ond setters.
```
Comment (pvojtechovsky) [inline src/main/java/spoon/support/DerivedProperty.java]: ond -> and

File: src/main/java/spoon/support/DerivedProperty.java [hunk 3]
```diff
@@ -23,10 +23,16 @@
...
+ *
+ * Contracts:
+ * - A setter with @DerivedProperty never triggers a model intercession event.
```
Comment (pvojtechovsky) [inline src/main/java/spoon/support/DerivedProperty.java]: > intercession event

Do you mean something like this?
```java
class CtConstructorCallImpl
...
	@DerivedProperty
	public <T extends CtActualTypeContainer> T addActualTypeArgument(CtTypeReference<?> actualTypeArgument) {
		if (getExecutable() != null) {
			getExecutable().addActualTypeArgument(actualTypeArgument);
		}
		return (T) this;
	}
```
It triggers the event, but not directly on the `this` node, but on some delegate? Did I understood it well? Improve description?

Is there a test for setters which does not start with `set` prefix, but with add/remove?

**Comment by monperrus** [comment]
clarified contract?

**Pull request merged**